### PR TITLE
Korean future fix

### PR DESCRIPTION
**Pull request by cecilebertin**
When I updated the combineSymbol function, I realized the future tense didn't work properly.
The if statement (if (stem[-1] !== 8), line 132) wasn't working correctly and I realized it was coded completely wrong. It pushed 8 to the array for all cases, and we ended up with arrays of 3 or 4 jamo's for latest character of the stem. 3 should be de maximum number of jamo's possible.
We were lucky it passed all the tests before, because the previous combineSymbol function didn't limit the length of the array.
I also realized we missed some cases for the future tense : when the stem ends with a ㅂ consonant, and the other verbs with a stem ending with a consonant other than ㄹ.

### CODE REVIEW

File: src/korean.js [hunk 1]
```diff
@@ -25,11 +25,11 @@ const breakdown = (input) => {
...
-  const initialValue = input[0] * 588;
-  const medialValue = input[1] * 28;
-  const finalValue = input[2] ? input[2] : 0;
-  const total = initialValue + medialValue + finalValue + 44032;
-  const finalWord = String.fromCharCode(total);
+  let unicodeTotal = (input[0] * 588) + (input[1] * 28) + 44032;
+  if(input.length === 3) {
+    unicodeTotal += input[2];
+  }
+  const finalWord = String.fromCharCode(unicodeTotal);
```
Comment (minsooshin) [inline src/korean.js]: can we simply return `String.fromCharCode(unicodeTotal)` without assigning?

Comment (cecilebertin) [reply src/korean.js]: @minsooshin you're right. I'll fix that right away.

**Comment by minsooshin** [comment]
@cecilebertin I don't know the history of this, but I think there is an issue on future tense. Could you take a look at https://github.com/llipio/conjugator/issues/41?

File: test/korean_tests.js [hunk 2]
```diff
@@ -107,28 +107,48 @@ describe('Korean', () => {
     });
  });
  describe('Future Tense', () => {
-   it('should conjugate regular verbs correctly', () => {
+   it('should conjugate verbs with stem ending with a vowel correctly', () => {
      const kc = new Korean();
-     let presentWord = kc.conjugate('하다', {tense: 'future'});
-     expect(presentWord).to.equal('할꺼야');
+     let futureWord = kc.conjugate('하다', {tense: 'future'});
+     expect(futureWord).to.equal('할꺼야');
```
Comment (minsooshin) [inline test/korean_tests.js]: I think it should be `할 거야`. `할꺼야` is pronunciation.

**Comment by cecilebertin** [comment]
@minsooshin ,  yes I just saw that. I'll take a look at it this afternoon with @yjlim5

File: test/korean_tests.js (hunk 2 above)
Comment (cecilebertin) [reply test/korean_tests.js]: @minsooshin Thanks! I'll add a fix for this too.

**Comment by minsooshin** [comment]
@cecilebertin thanks!!!

File: test/korean_tests.js (hunk 2 above)
Comment (cecilebertin) [reply test/korean_tests.js]: @minsooshin I looked it up, http://www.koreanwikiproject.com/wiki/A/V_%2B_(으)ㄹ_거예요
I think there should be a space in between for all the cases.

Comment (minsooshin) [reply test/korean_tests.js]: @cecilebertin that's right. Also, the example you provide is to senior people (Informal polite). So, for your test cases, I think `To make the low form (반말), the 예요 part becomes 야: A/V+(으)ㄹ 거야` is the correct reference. That's what I said before; it should be `할 거야`. There is a whitespace between `할` and `거야`.

File: src/korean.js [hunk 3]
```diff
@@ -129,10 +128,18 @@ class Korean {
 
   doFuture (word) {
     const stem = breakdown(word.slice(0, -1));
-    if (stem[-1] !== 8) {
+    if (stem.length < 3) {
       stem.push(8);
+      return `${combineSymbols(stem)} 꺼야`;
```
Comment (minsooshin) [inline src/korean.js]:
```js
return `${combineSymbols(stem)} 거야`;
```

not `꺼야`

Comment (minsooshin) [reply src/korean.js]: `꺼야` is written the pronunciation.

Comment (cecilebertin) [reply src/korean.js]: @minsooshin I think you're right again :) ,I looked it up. Nice catch!
I'll add another commit to fix this.

Comment (minsooshin) [reply src/korean.js]: @cecilebertin thanks a lot for quick fix and feedback!!!

Comment (cecilebertin) [reply src/korean.js]: @minsooshin thanks for the thorough review!!!

**Review by yjlim5 — APPROVED** [review]

**Comment by yjlim5** [comment]
Thanks for catching the details and fixing them @cecilebertin and @minsooshin.  Great team work!

**Pull request merged**